champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/warlock/axe/Clone.java

```diff
@@ -0,0 +1,303 @@
+        UtilPlayer.slowHealth(champions, player, -proposedHealth, 5, false);
+
+        Vindicator clone = (Vindicator) player.getWorld().spawnEntity(player.getLocation(), EntityType.VINDICATOR);
+
+        Disguise disguise = new PlayerDisguise(player).setNameVisible(false);
+        DisguiseAPI.disguiseToAll(clone, disguise);
+
+        setCloneProperties(clone, player);
+
+        //leap the clone forward
+        VelocityData velocityData = new VelocityData(clone.getLocation().getDirection(), leapStrength, false, 0.0D, 0.2D, 1.0D, false);
+        UtilVelocity.velocity(clone, player, velocityData, VelocityType.CUSTOM);
+
+        //Find nearby enemies relative to the clones location after teleporting
+        List<Player> nearbyEnemies = UtilPlayer.getNearbyEnemies(player, clone.getLocation(), 24);
+        Player initTarget = null;
+        if (!nearbyEnemies.isEmpty()) {
+            //Pick a random nearby enemy
+            Random random = new Random();
+            initTarget = nearbyEnemies.get(random.nextInt(nearbyEnemies.size()));
+        }
+
+        clones.put(clone, System.currentTimeMillis());
+        Bukkit.getMobGoals().addGoal(clone, 0, new ClonePathFinder(champions, clone, player, initTarget));
+    }
+
+    @UpdateEvent(delay = 100)
+    public void onUpdate() {
+        Iterator<Map.Entry<Vindicator, Long>> it = clones.entrySet().iterator();
+        while (it.hasNext()) {
+            Vindicator clone = it.next().getKey();
+
+            if (clone == null) {
+                it.remove();
+                continue;
+            }
+
+            final Player player = getCloneOwner(clone);
+
+            if (player == null || !player.isOnline()) {
+                it.remove();
+                removeClone(clone);
+                continue;
+            }
+
+            int level = getLevel(player);
+
+            if (!(level > 0)) {
+                it.remove();
+                removeClone(clone);
+                continue;
+            }
+
+            if (UtilTime.elapsed(clones.get(clone), (long) getDuration(level) * 1000)) {
+                it.remove();
+                removeClone(clone);
+            }
+        }
+    }
+
+    @EventHandler
+    public void onEntityTarget(EntityTargetEvent event) {
+        if (event.getEntity() instanceof Vindicator clone && event.getTarget() instanceof Player target && clones.containsKey(clone)) {
+            final Player player = getCloneOwner(clone);
+            if (UtilEntity.getRelation(player, target) == EntityProperty.FRIENDLY) {
+                event.setCancelled(true);
+            }
+        }
+    }
+
+    @EventHandler
+    public void onCustomDamageEvent(CustomDamageEvent event) {
+        if (event.getDamagee() instanceof Player player && event.getDamager() instanceof Vindicator clone && clones.containsKey(clone)) {
+            event.setCancelled(true);
+            handleCloneDamage(player, clone);
```

Comment:
Why cancel the event if you're just gonna call it again in handleCloneDamage?

You should just be able to override the properties here